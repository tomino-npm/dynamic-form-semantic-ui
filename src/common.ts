import { FormElement, DataSet } from '@tomino/dynamic-form';
import { initCss } from '@tomino/toolbelt';

export type FormControlProps = {
  formControl: FormElement;
  owner: DataSet;
  handlers?: { [index: string]: (...params: any[]) => any };
};

export const { css, renderCss } = initCss('d-form');
