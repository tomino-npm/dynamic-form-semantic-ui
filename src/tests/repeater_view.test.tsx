import * as React from 'react';
import * as renderer from 'react-test-renderer';

import { create } from './form_query_data';
import { JSONSchema, FormDefinition, FormModel } from '@tomino/dynamic-form';
import { TestComponent } from './common';

describe('Form', () => {
  const schema: JSONSchema = {
    type: 'object',
    properties: {
      countries: {
        type: 'array',
        items: {
          type: 'object',
          properties: {
            id: {
              type: 'string'
            },
            name: {
              type: 'string'
            },
            capital: {
              type: 'string'
            }
          }
        }
      }
    }
  };

  const controlData = {
    countries: [
      { name: 'Slovakia', capital: 'Bratislava' },
      { name: 'Australia', capital: 'Canberra' }
    ]
  };

  const formDefinition: FormDefinition = create.form({
    elements: [
      create.formElement({
        row: 0,
        column: 0,
        width: 16,
        control: 'Repeater',
        source: 'countries',
        elements: [
          create.formElement({
            row: 0,
            column: 0,
            width: 8,
            control: 'Input',
            label: 'Name',
            source: 'name'
          }),
          create.formElement({
            row: 0,
            column: 8,
            width: 8,
            control: 'Input',
            label: 'Capital',
            source: 'capital'
          }),
          create.formElement({
            row: 0,
            column: 15,
            width: 1,
            control: 'DeleteButton',
            label: '\xa0'
          })
        ]
      })
    ]
  });

  describe('Repeater', () => {
    function componentWithData() {
      const form = new FormModel(formDefinition, schema, controlData);

      // just another notation
      return <TestComponent form={form} />;
    }

    it('renders correctly', () => {
      const comp = renderer.create(componentWithData());
      expect(comp).toMatchSnapshot();
    });

    return {
      componentWithData
    };

    // it('changes value and all related formulas', () => {
    //   // const component = renderer.create(data.component);
    //   // const root = component.root;
    //   // const agree = root.findByProps({ name: 'agree' });
    //   // agree.props.onChange(null, { value: false });
    //   // const disagree = root.findByProps({ name: 'disagree' });
    //   // disagree.props.onChange(null, { value: true });
    //   // expect(component).toMatchSnapshot();
    // });
  });
});
