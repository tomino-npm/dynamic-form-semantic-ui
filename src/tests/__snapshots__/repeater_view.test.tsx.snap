// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Repeater renders correctly 1`] = `
<div
  className="ui segment ui form"
>
  <style>
    .d-form-bgcibidbjc { 
  color: pink;
  font-size: 100px !important;
 }
.d-form-Aebfdiadfh { 
  padding: 9px 0px;
 }
.d-form-Adiagibjfa { 
  margin-right: 12px;
  margin-bottom: 6px;
 }
.d-form-bidfabidci { 
  margin: 6px 0px 12px 0px;
 }
.d-form-Abdgbhfjheg { 
  margin-bottom: 0px !important;
 }

  .fieldSet-bejaheefjb {
    border: solid 1px #ddd;
    border-radius: 5px;
    background-color: #fafafa;
  }


  .fieldSet-bejaheefjb legend {
    padding: 0px 6px;
    font-weight: bold;
  }
  </style>
  <div
    className="ui form"
  >
    <div
      className="fields"
    >
      <div
        className="sixteen wide field"
      >
        <fieldset
          className=".fieldSet-bejaheefjb"
        >
          <style>
            .d-form-bgcibidbjc { 
  color: pink;
  font-size: 100px !important;
 }
.d-form-Aebfdiadfh { 
  padding: 9px 0px;
 }
.d-form-Adiagibjfa { 
  margin-right: 12px;
  margin-bottom: 6px;
 }
.d-form-bidfabidci { 
  margin: 6px 0px 12px 0px;
 }
.d-form-Abdgbhfjheg { 
  margin-bottom: 0px !important;
 }

  .fieldSet-bejaheefjb {
    border: solid 1px #ddd;
    border-radius: 5px;
    background-color: #fafafa;
  }


  .fieldSet-bejaheefjb legend {
    padding: 0px 6px;
    font-weight: bold;
  }
          </style>
          <div
            className="ui form"
          >
            <div
              className="fields"
            >
              <div
                className="eight wide field"
              >
                <label
                  htmlFor="name"
                >
                  Name
                </label>
                <div
                  className="ui input"
                >
                  <input
                    name="name"
                    onChange={[Function]}
                    type="text"
                    value="Slovakia"
                  />
                </div>
              </div>
              <div
                className="eight wide field"
              >
                <label
                  htmlFor="capital"
                >
                  Capital
                </label>
                <div
                  className="ui input"
                >
                  <input
                    name="capital"
                    onChange={[Function]}
                    type="text"
                    value="Bratislava"
                  />
                </div>
              </div>
              <div
                className="one wide field"
              >
                <label>
                   
                </label>
                <button
                  className="ui red icon button"
                  onClick={[Function]}
                  role="button"
                >
                  <i
                    aria-hidden="true"
                    className="trash icon"
                  />
                </button>
              </div>
            </div>
          </div>
          <style>
            .d-form-bgcibidbjc { 
  color: pink;
  font-size: 100px !important;
 }
.d-form-Aebfdiadfh { 
  padding: 9px 0px;
 }
.d-form-Adiagibjfa { 
  margin-right: 12px;
  margin-bottom: 6px;
 }
.d-form-bidfabidci { 
  margin: 6px 0px 12px 0px;
 }
.d-form-Abdgbhfjheg { 
  margin-bottom: 0px !important;
 }

  .fieldSet-bejaheefjb {
    border: solid 1px #ddd;
    border-radius: 5px;
    background-color: #fafafa;
  }


  .fieldSet-bejaheefjb legend {
    padding: 0px 6px;
    font-weight: bold;
  }
          </style>
          <div
            className="ui form"
          >
            <div
              className="fields"
            >
              <div
                className="eight wide field"
              >
                <label
                  htmlFor="name"
                >
                  Name
                </label>
                <div
                  className="ui input"
                >
                  <input
                    name="name"
                    onChange={[Function]}
                    type="text"
                    value="Australia"
                  />
                </div>
              </div>
              <div
                className="eight wide field"
              >
                <label
                  htmlFor="capital"
                >
                  Capital
                </label>
                <div
                  className="ui input"
                >
                  <input
                    name="capital"
                    onChange={[Function]}
                    type="text"
                    value="Canberra"
                  />
                </div>
              </div>
              <div
                className="one wide field"
              >
                <label>
                   
                </label>
                <button
                  className="ui red icon button"
                  onClick={[Function]}
                  role="button"
                >
                  <i
                    aria-hidden="true"
                    className="trash icon"
                  />
                </button>
              </div>
            </div>
          </div>
          <button
            className="ui icon primary left labeled button"
            onClick={[Function]}
            role="button"
          >
            <i
              aria-hidden="true"
              className="plus icon"
            />
            Add
          </button>
        </fieldset>
      </div>
    </div>
  </div>
</div>
`;
